modROS.lua

```diff
@@ -282,32 +282,28 @@ end
 
 function ModROS:publish_laser_scan_func()
     local radius = 0.2
-    local x, y, z = getWorldTranslation(self.instance_veh.cameraNode)
-
     -- cache locally
     local cos = math.cos
     local sin = math.sin
     local LS_FOV = mod_config.laser_scan.angle_max - mod_config.laser_scan.angle_min
 
     -- calculate nr of steps between rays
     local delta_theta = LS_FOV / (mod_config.laser_scan.num_rays - 1)
-    -- the "delta_y_total = 0.05" varaible is the total distance from the z axis of non-tilted laser scan layer to the z axis of last laser scan layer (most tilted one)
-    -- and is to determine how much the laser scan layers are tilted, the larger the length is, the more titlted the laser scan layer is
-    -- And "delta_y" is added between the scanning planes, along +y direction from the origin of the laser scanner
-    local delta_y_total = 0.05
-    -- quarter of length as there are four segements between 5 laser scan layers
-    local delta_y = delta_y_total / (mod_config.laser_scan.num_layers - 1)
     for i = 0, mod_config.laser_scan.num_layers-1 do
         self.laser_scan_array = {}
+        -- get the cooridante(world) of each laser scanner's origin
+        -- "mod_config.laser_scan.inter_layer_distance" is added between the scanning planes, along +y direction from the lowest laser scan plane
+        -- and all laser scan planes are parallel to each other
+        local orig_x, orig_y, orig_z = localToWorld(self.instance_veh.cameraNode, 0, mod_config.laser_scan.inter_layer_distance * i, 0)
```

Comment:
```suggestion
        local laser_dy = mod_config.laser_scan.inter_layer_distance * i
        local orig_x, orig_y, orig_z = localToWorld(self.instance_veh.cameraNode, 0, laser_dy, 0)
```

could then use `laser_dy` later on when setting up `tf_base_link_laser_frame_i` as well.

(name could be different. Perhaps something else would make more sense)